src/assets/scripts/client/ui/TutorialView.js

```diff
@@ -240,9 +243,64 @@ export default class TutorialView {
         this.tutorial.steps = [];
         this.tutorial.step = 0;
 
-        const tutorial_position = [0.1, 0.85];
-        const departureAircraft = prop.aircraft.list.filter((aircraftModel) => aircraftModel.isDeparture())[0];
-
+        // these always get evaluated on-the-fly.
+        this.tutorial.liverefs = {
+            airport: function() {
+                return AirportController.airport_get();
+            },
+            departureAircraft: function() {
+                return prop.aircraft.list.filter((aircraftModel) => aircraftModel.isDeparture())[0];
+            }
+        };
+
+        zlsa.atc.loadAsset({ url: 'assets/tutorial/tutorial.json', immediate: true })
+            .done((response) => {
+                response.forEach((step) => {
+                    if (Array.isArray(step.replace)) {
```

Comment:
could we abstract this logic into a dedicated helper or service?  there is quite a bit here that isn't dedicated to the TutorialView, but parsing json.

Reply:
moved handling of each step into a separate function.